CANTroller2/src/devices.h

```diff
@@ -318,6 +311,14 @@ class Transducer : public Device {
         return false;
     }
 
+    // Convert units from base numerical value to disp units:  val_native = m-factor*val_numeric + offset  -or-  val_native = m-factor/val_numeric + offset  where m-factor, b-offset, invert are set here
+    void set_convert(float arg_m_factor, float arg_b_offset, bool arg_invert) {
+        _m_factor = arg_m_factor;
+        _b_offset = arg_b_offset;
+        _invert = arg_invert;
+        set_native(native.get()); // NOTE: should we really force a conversion when the factors update? any reason not to?
```

Comment:
Resolve this comment. fwiw I think getting the value should be separate from processing it. This function seems to be setting flags/variables to be used for processing

Reply:
It _is_ separate. here we are changing the conversion values, and the question is whether to immediately do a conversion. The thought was that we would get the current native value and convert it to the new human-readable value based on the new conversion factors. set_native() is designed to convert to human automatically - that is, do the equivalent of human.set(to_human(native.get()). but i kind of agree that we shouldn't make that kind of assumption here (what if the input is going from human->native instead, as is the case with PWMs?). I'll change it.